nmigen_boards/orangecrab_r0d2.py

```diff
@@ -0,0 +1,114 @@
+import os
+import subprocess
+import shutil
+
+from nmigen.build import *
+from nmigen.vendor.lattice_ecp5 import *
+from .resources import *
+
+
+__all__ = ["OrangeCrabR0D2Platform"]
+
+
+class OrangeCrabR0D2Platform(LatticeECP5Platform):
+    device      = "LFE5U-25F"
+    package     = "MG285"
+    speed       = "8"
+    default_clk = "clk_48MHz"
+    resources   = [
+        Resource("clk_48MHz", 0, Pins("A9", dir="i"),
+                 Clock(48e6), Attrs(IO_STANDARD="LVCMOS33")),
+        Resource("rst_n", 0, Pins("V17"), Attrs(IO_TYPE="LVCMOS33")),
```

Comment:
`rst`, `PinsN`

Reply:
Also swapped to an output after discussing with Greg that it's actually for toggling PROGRAMN from gateware

Reply:
A `programn` pin should be *definitely* not called `"rst"` or anything like it; that would be extremely misleading!

Reply:
yeah I nearly added it to default_rst